tests/backup_helper.go

```diff
@@ -10164,3 +10165,93 @@ func GetApplicationSpecForFeature(feature string) ([]string, error) {
 	}
 	return nil, fmt.Errorf("feature %s not found", feature)
 }
+
+// FindProvisionerToSnapshotClassByNamespace finds all volumes in the specified namespace, retrieves their provisioners, and maps them to their corresponding VolumeSnapshotClass.
+func FindProvisionerToSnapshotClassByNamespace(namespace string) (map[string]string, error) {
+	// Create a map to store the provisioner and corresponding VolumeSnapshotClass
+	provisionerToSnapshotClassMap := make(map[string]string)
+	// Get the list of volumes in the specified namespace
+	pvcList, err := k8sCore.GetPersistentVolumeClaims(namespace, nil)
+	if err != nil {
+		return nil, err
+	}
+	// Get the list of VolumeSnapshotClasses
+	snapshotClasses, err := Inst().S.GetAllSnapshotClasses()
+	if err != nil {
+		return nil, err
+	}
+	for _, volume := range pvcList.Items {
+		scProvisioner, err := k8sCore.GetStorageProvisionerForPVC(&volume)
+		if err != nil {
+			return nil, err
+		}
+		for _, snapshotClass := range snapshotClasses.Items {
+			if snapshotClass.Driver == scProvisioner {
+				provisionerToSnapshotClassMap[scProvisioner] = snapshotClass.Name
+			}
+		}
```

Comment:
We can avoid this nested loop by creating a map of scProvisioners and have emptry string initially. Then in the second loop, we can just do
```
provisionerToSnapshotClassMap[scProvisioner] = snapshotClass.Name
```

Reply:
As suggested changed this to FindProvisionerToSnapshotClassMap , which will fetch map at cluster level.